### ISSUE TITLE
Zmodem Transfers

### ISSUE THREAD

**Issue by leenooks** [FozzTexx/tcpser#15]
Howdy, do you have any suggestions for getting tcpser to work with zmodem transfers?

I have my BBS running in qemu (i386 emulation) with serial ports (and MSDOS).

When I telnet to the host, and interact as a user, while the experience is slow (like serial port speeds), things appear OK.

However, when I have an FTN mailer send mail, the Zmodem transfers always fail. I've wondered if its because of LAN speeds actually being too fast, and there is too much traffic in queue (and thus sent be the sending system) to process before the receiving system can send ACKs. 

Its probably also QEMU dropping data, because its coming too fast, so Zmodem CRCs fail (I see a lot of that as well)...

I'm wondering if there is a way to force TCPser to slow down if that would help?

**Comment by FozzTexx** [comment]
Zmodem requires an 8-bit clean connection. Your FTN mailer is probably not negotiating an 8-bit connection, or your BBS is using parity and tcpser is not able to do 8-bit.

**Comment by leenooks** [comment]
So is there a way to force tpcser to only do 8bit, or it cannot do it at all?

**Comment by FozzTexx** [comment]
The problem isn't with tcpser, it's with whatever you have connecting to it.

**Comment by leenooks** [comment]
Hmm.. so I'm not sure that tcpser will work for me then? I've tried:
* QEMU with -serial tcp:... and tcpser with -v
* QEMU with -serial /dev/tnt0, and tty0tty0 and tcpser -d /dev/tnt1

And both work for interactive sessions, but not for zmodem (mailer) transfers. Unfortunately in the first case, using tcpser with -v, the mailer thinks that the DCD is high.

I will try a direct zmodem transfer in DOS, to szrz in the host, using the tty0tty devices and see if that works, which I guess will confirm if its the IP side or the serial port side...

**Comment by leenooks** [comment]
So I took the mailer out of the question, and tested with TERMINATE.

I used a config with QEMU, using tty0tty to connect QEMU to tcpser or minicom.

When I uploaded from terminate to minicom, it was lightning fast and worked without any issues.

When I uploaded from terminate to my lan bbs through TCPser, it bombed - lots of transmit errors and retries.

When I uploaded from minicom to terminate, it too was problematic, but it was way faster and progressed further than using tcpser.

So I think I have more problems then just TCPser - and its probably not an easy one to diagnose.
:(
